vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java

```diff
@@ -1,85 +1,32 @@
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
-import edu.asu.diging.vspace.core.data.ExternalLinkDisplayRepository;
-import edu.asu.diging.vspace.core.data.ExternalLinkRepository;
 import edu.asu.diging.vspace.core.data.ImageRepository;
-import edu.asu.diging.vspace.core.data.ModuleLinkRepository;
-import edu.asu.diging.vspace.core.data.SpaceLinkRepository;
-import edu.asu.diging.vspace.core.data.display.ModuleLinkDisplayRepository;
-import edu.asu.diging.vspace.core.data.display.SpaceLinkDisplayRepository;
 import edu.asu.diging.vspace.core.exception.FileStorageException;
 import edu.asu.diging.vspace.core.exception.ImageCouldNotBeStoredException;
+import edu.asu.diging.vspace.core.exception.LinkDoesNotExistsException;
 import edu.asu.diging.vspace.core.exception.SpaceDoesNotExistException;
-import edu.asu.diging.vspace.core.factory.IExternalLinkDisplayFactory;
-import edu.asu.diging.vspace.core.factory.IExternalLinkFactory;
 import edu.asu.diging.vspace.core.factory.IImageFactory;
-import edu.asu.diging.vspace.core.factory.IModuleLinkDisplayFactory;
-import edu.asu.diging.vspace.core.factory.IModuleLinkFactory;
-import edu.asu.diging.vspace.core.factory.ISpaceLinkDisplayFactory;
-import edu.asu.diging.vspace.core.factory.ISpaceLinkFactory;
 import edu.asu.diging.vspace.core.file.IStorageEngine;
-import edu.asu.diging.vspace.core.model.IExternalLink;
-import edu.asu.diging.vspace.core.model.IModule;
-import edu.asu.diging.vspace.core.model.IModuleLink;
+import edu.asu.diging.vspace.core.model.ILink;
 import edu.asu.diging.vspace.core.model.ISpace;
-import edu.asu.diging.vspace.core.model.ISpaceLink;
 import edu.asu.diging.vspace.core.model.IVSImage;
+import edu.asu.diging.vspace.core.model.IVSpaceElement;
 import edu.asu.diging.vspace.core.model.display.DisplayType;
-import edu.asu.diging.vspace.core.model.display.IExternalLinkDisplay;
-import edu.asu.diging.vspace.core.model.display.IModuleLinkDisplay;
-import edu.asu.diging.vspace.core.model.display.ISpaceLinkDisplay;
-import edu.asu.diging.vspace.core.model.display.impl.ExternalLinkDisplay;
-import edu.asu.diging.vspace.core.model.display.impl.ModuleLinkDisplay;
-import edu.asu.diging.vspace.core.model.display.impl.SpaceLinkDisplay;
-import edu.asu.diging.vspace.core.model.impl.ExternalLink;
-import edu.asu.diging.vspace.core.model.impl.ModuleLink;
-import edu.asu.diging.vspace.core.model.impl.SpaceLink;
+import edu.asu.diging.vspace.core.model.display.ILinkDisplay;
 import edu.asu.diging.vspace.core.model.impl.VSImage;
 import edu.asu.diging.vspace.core.services.ILinkManager;
-import edu.asu.diging.vspace.core.services.IModuleManager;
 import edu.asu.diging.vspace.core.services.ISpaceManager;
 
 @Transactional
-@Service
-public class LinkManager implements ILinkManager {
+public abstract class LinkManager<L,T, U> implements ILinkManager<L,T,U>{
```

Comment:
2 things: 
- I don't think you need L,T, U in the class definition. should be enough in the interface I believe.
- if you make `L extends ILink` you don't need to cast L e.g. in line 45